go/vt/vtgate/planbuilder/dml_planner.go

```diff
@@ -42,7 +43,22 @@ func rewriteRoutedTables(stmt sqlparser.Statement, vschema plancontext.VSchema)
 			return false, vterrors.VT09014()
 		}
 
-		if vschemaTable.Name.String() != tableName.Name.String() {
+		if vschemaTable.Type == vindexes.TypeReference && vschemaTable.Source != nil {
+			// This is a reference table. We need to rewrite the table name and keyspace to that of the source table.
+			vschemaTable, vindexTbl, _, _, _, err = vschema.FindTableOrVindex(vschemaTable.Source.TableName)
+			if err != nil {
+				return false, err
+			}
+			if vindexTbl != nil {
+				// vindex cannot be present in a dml statement.
+				return false, vterrors.VT09014()
+			}
+			tableName.Qualifier = vschemaTable.GetTableName().Qualifier
+			tableName.Name = sqlparser.NewIdentifierCS(vschemaTable.Name.String())
+			newAliasTbl := sqlparser.NewAliasedTableExpr(tableName, "")
+			aliasTbl.Expr = newAliasTbl.Expr
+			aliasTbl.As = newAliasTbl.As
```

Comment:
The alias should be either an existing one or the reference table name. Should not be source table name.

Reply:
better to add some tests around it in `dml_cases.json`

Reply:
I see you have created a new test file for it. So, those can be added there.
DML with alias.

Reply:
Moved the test cases into reference_cases.json.

Reply:
> The alias should be either an existing one or the reference table name. Should not be source table name.

So I should only change the qualifier and retain the existing alias' table name?

Reply:
@harshit-gangal, updated the code as we discussed and added more test cases for queries where aliases are specified.